plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/modules/SourceModule.java

```diff
@@ -694,10 +696,38 @@ public TokensList getCompletionsForBase(ICompletionState initialState, ICodeComp
             state = initialState.getCopy();
             state.setActivationToken(s);
             return manager.getCompletionsForModule(this, state);
+        } else if (classDef.bases[baseIndex] instanceof Subscript) {
+            TokensList tokens = new TokensList();
+            Subscript subscript = (Subscript) classDef.bases[baseIndex];
+            tokens.addAll(getCompletionsForValueAsName(manager, initialState, subscript.value));
+            if (subscript.slice instanceof Index) {
+                Index index = (Index) subscript.slice;
+                tokens.addAll(getCompletionsForValueAsName(manager, initialState, index.value));
+            }
+            if (tokens.size() > 0) {
+                return tokens;
+            }
         }
         return null;
     }
 
+    private TokensList getCompletionsForValueAsName(ICodeCompletionASTManager manager, ICompletionState state,
+            exprType value) throws CompletionRecursionException {
+        if (value instanceof Name) {
```

Comment:
Instead of checking whether it's a `Name`, you probably want to use: `org.python.pydev.parser.visitors.NodeUtils.getFullRepresentationString(SimpleNode, boolean)`.